### PR TITLE
Needs decrease & bugfix

### PR DESCRIPTION
**Pull request by SamuelllG**
implemented needs decrease over time depending on moving, long moving, fighting

refactored SanityController: pull out IsFighting to AttackLogic

fix weapon damage bug for enemies

Note: I don't like how AttackLogic is used for both enemies and player at all. There is some dirty stuff in it with the Hit method etc as stated in another comment. Unfortunately I won't have the time to do a refactor (i.e. basically a whole rework) of the attack logic... quite a few things depend on the AttackLogic.

### CODE REVIEW

**Comment by Kaes3kuch3n** [comment]
Merge conflicts 😅

File: Assets/Scripts/Entity/AttackLogic.cs [hunk 1]
```diff
@@ -135,9 +178,11 @@ private void HitFinished()
             if (Status == AttackStatus.Hit)
             {
                 // deal damage
-                // Add damage boost from weapon
-                int boostedDamage = damage + InventoryManager.Instance.DamageBoostFromEquipable;
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Entity/AttackLogic.cs:139]: Lol wtf 😂 Was this applied to every entity?

File: Assets/Scripts/Entity/Player/NeedsController.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                }
+            }
+            // don't apply fight malus, when running away (in a fight)
+            // just apply if really fighting (mind that IsFighting is true a certain time after having killed an enemy)
+            else if (_attackLogic.IsFighting)
+            {
+                scaledSaturationTick *= fightMalusSaturation;
+                scaledHydrationTick *= fightMalusHydration;
+            }
+            Debug.Log(scaledSaturationTick);
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Entity/Player/NeedsController.cs]: Please remove debug statements :)

**Review by Kaes3kuch3n — CHANGES_REQUESTED** [review]
Yeah, not very clean 😅 But we don't have enough time left to fix all this, I guess 😬

File: Assets/Scripts/Entity/AttackLogic.cs (hunk 1 above)
Comment (DavidPgl) [reply Assets/Scripts/Entity/AttackLogic.cs:139]: Who would add something like that... He must be very smart 😅  * *hust* *

**Review by DavidPgl — DISMISSED** [review]

File: Assets/Scripts/Entity/Player/NeedsController.cs (hunk 2 above)
Comment (SamuelllG) [reply Assets/Scripts/Entity/Player/NeedsController.cs]: whups

File: Assets/Scripts/Entity/AttackLogic.cs [hunk 3]
```diff
@@ -182,7 +182,11 @@ private void HitFinished()
                 // add damage boost from weapon if is player
                 if (_enemyController == null)
                     damageDealt += InventoryManager.Instance.DamageBoostFromEquipable;
-                Target.Hit(damageDealt, _enemyController);
+                Target.Hit(damageDealt, out int targetHealth, _enemyController);
+                if (targetHealth <= 0)
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Entity/AttackLogic.cs:186]: I guess this shouldn't be called twice 🤔

**Review by Kaes3kuch3n — CHANGES_REQUESTED** [review]
Sorry, one more thing 😅 Might come from merging master 🤔

File: Assets/Scripts/Entity/AttackLogic.cs (hunk 3 above)
Comment (SamuelllG) [reply Assets/Scripts/Entity/AttackLogic.cs:186]: Oh dear, sorry 🤦‍♂

**Review by thuemmels — APPROVED** [review]

**Review by Kaes3kuch3n — APPROVED** [review]

**Pull request merged**